### ISSUE TITLE
force-media-title with mpv not working

### ISSUE THREAD

**Issue by crackerbear** [streamlink/streamlink#3036]
## Bug Report

- [x] This is a bug report and I have read the contribution guidelines.

### Description

This is very likely my own stupidity rather than being a bug, but I'm trying to [replicate commands](https://github.com/streamlink/streamlink-twitch-gui/blob/fa310904daf46cfa5733d84c34968ca8bcf0d751/src/config/players.json) that Streamlink Twitch GUI uses, in particular ```force-media-title="{name} - {game} - {status}"``` with mpv to display the correct the title of a Twitch stream. 

### Expected / Actual behavior

Expected: With the GUI and [Set custom media title enabled](https://i.imgur.com/drAL7U9.png), the media title is [correctly displayed](https://i.imgur.com/wBXGLaL.jpg). 

Actual behavior: With the CLI and using the same command of ```force-media-title="{name} - {game} - {status}"```, mpv [displays it literally](https://i.imgur.com/uL9lXTS.jpg) without fetching the information.

### Reproduction steps / Explicit stream URLs to test

1. In %APPDATA%\streamlink\streamlinkrc: ```"C:\Program Files\mpv\mpv.exe" --keep-open=yes --force-seekable=yes --hr-seek=yes --hr-seek-framedrop=yes --force-media-title="{name} - {game} - {status}"```
2. In cmd: ```streamlink --loglevel debug --twitch-disable-reruns --twitch-disable-ads --twitch-disable-hosting --twitch-low-latency --title "{author} - {title} - {category}" https://www.twitch.tv/summit1g 720p```

### Log output

```
streamlink --loglevel debug --twitch-disable-reruns --twitch-disable-ads --twitch-disable-hosting --twitch-low-latency --title "{author} - {title} - {category}" https://www.twitch.tv/summit1g 720p
[cli][debug] OS:         Windows 10
[cli][debug] Python:     3.6.6
[cli][debug] Streamlink: 1.4.1
[cli][debug] Requests(2.23.0), Socks(1.7.1), Websocket(0.57.0)
[cli][info] Found matching plugin twitch for URL https://www.twitch.tv/summit1g
[cli][debug] Plugin specific arguments:
[cli][debug]  --twitch-disable-hosting=True (disable_hosting)
[cli][debug]  --twitch-disable-ads=True (disable_ads)
[cli][debug]  --twitch-disable-reruns=True (disable_reruns)
[cli][debug]  --twitch-low-latency=True (low_latency)
[plugin.twitch][debug] Getting live HLS streams for summit1g
[utils.l10n][debug] Language code: en_GB
[cli][info] Available streams: audio_only, 160p (worst), 360p, 480p, 720p, 720p60, 1080p60 (best)
[cli][info] Opening stream: 720p (hls)
[plugin.twitch][info] Will skip ad segments
[plugin.twitch][info] Low latency streaming (HLS live edge: 2)
[stream.hls][debug] Reloading playlist
[stream.hls][debug] First Sequence: 8431; Last Sequence: 8439
[stream.hls][debug] Start offset: 0; Duration: None; Start Sequence: 8438; End Sequence: None
[stream.hls][debug] Adding segment 8438 to queue
[cli][debug] Pre-buffering 8192 bytes
[stream.hls][debug] Adding segment 8439 to queue
[cli][info] Starting player: "C:\Program Files\mpv\mpv.exe" --keep-open=yes --force-seekable=yes --hr-seek=yes --hr-seek-framedrop=yes --force-media-title="{name} - {game} - {status}"
[cli.output][debug] Opening subprocess: "C:\Program Files\mpv\mpv.exe" --keep-open=yes --force-seekable=yes --hr-seek=yes --hr-seek-framedrop=yes --force-media-title="{name} - {game} - {status}" "--title=summit1g - gta chillin  [ @summit1g ] - Grand Theft Auto V" -
[cli][debug] Writing stream to output
```

### Additional comments, screenshots, etc.

Is it to do with escaping the command or am I missing something else completely?

**Comment by bastimeyer** [comment]
> I'm trying to replicate commands that Streamlink Twitch GUI uses

Streamlink Twitch GUI [has its own substitution system for curly bracket variables](https://github.com/streamlink/streamlink-twitch-gui/wiki/Player-configuration#custom-player-parameters). This works independently of Streamlink and was implemented at least three years before Streamlink's `--title` parameter was added (#1576) where automated custom titles became available.

If you take a look at the [`--title` documentation](https://streamlink.github.io/cli.html#cmdoption-t), you'll see that only the `title`, `author`, `category`, `game`, and `url` variables are supported - if a plugin supports them - and that they have to be set inside the `--title` parameter.

The usage of variables other than `{filename}` inside `--player-args` (or `--player`), or unrecognized variables like `{name}` or `{status}` inside `--title` is thus invalid. This has also nothing to do with MPV's `--force-media-title`, as it doesn't care where those variables are set.

> force-media-title with mpv not working

Streamlink doesn't set this MPV parameter when applying the title value and [it only sets MPV's `--title` param](https://github.com/streamlink/streamlink/blob/1.4.1/src/streamlink_cli/output.py#L190-L194). This means that you can't set a variable value for that player parameter.

The Twitch GUI sets [both MPV params for customizing the title](https://github.com/streamlink/streamlink-twitch-gui/blob/v1.10.0/src/config/players.json#L94-L107), but again, this works completely independently.

**Comment by crackerbear** [comment]
> The Twitch GUI sets [both MPV params for customizing the title](https://github.com/streamlink/streamlink-twitch-gui/blob/v1.10.0/src/config/players.json#L94-L107), but again, this works completely independently.

So in layman's terms, Twitch GUI has it's own code that sets MPV's media title that cannot be replicated in Streamlink CLI on it's own?

**Comment by bastimeyer** [comment]
Yes. If you want, you can submit a PR with the addition of the `--force-media-title` parameter here:
https://github.com/streamlink/streamlink/blob/1.4.1/src/streamlink_cli/output.py#L194

Since I'm not a fan of the current implementation, a better fix for this problem would be having a `{title}` variable available in the `--player-args` / `--player` parameter ([in addition to `{filename}`](https://github.com/streamlink/streamlink/blob/1.4.1/src/streamlink_cli/output.py#L205)) which gets its value from the `--title` parameter (`self.title`).

Then the user could add a custom `--force-media-title` value for MPV like the following example, and even customize window/playback titles of unsupported players as well:

```bash
streamlink \
  --title '{author} - {title} - {category}' \
  --player mpv \
  --player-args '"--title={title}" "--force-media-title={title}" {filename}' \
  $URL \
  $STREAM
```

**Comment by crackerbear** [comment]
Your suggestion does seem good, however I wouldn't know how to word the PR (Pull Request?) in a way that it makes sense. If it wasn't completely obvious, I don't know much about coding, I naively thought I could do away with the GUI and use the stand-alone CLI alongside the GUI parameters.

Thanks for the information and help anyway!
